controller/src/main/java/org/jboss/as/controller/AbstractOperationStepHandler.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015, Red Hat, Inc., and individual contributors as indicated
+ * by the @authors tag.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.controller;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.jboss.dmr.ModelNode;
+
+/** Simple helper class to provide similar behaviour than AbstractAddStepHandler but for generic handlers
+ * @author Tomaz Cerar (c) 2015 Red Hat Inc.
+ */
+public abstract class AbstractOperationStepHandler implements OperationStepHandler {
+    private final List<AttributeDefinition> attributes;
+
+    public AbstractOperationStepHandler(AttributeDefinition... parameters) {
+        this.attributes = Arrays.asList(parameters);
+    }
+
+    public AbstractOperationStepHandler(OperationDefinition operationDefinition) {
+        this.attributes = Arrays.asList(operationDefinition.getParameters());
+    }
+
+    protected void populateModel(final ModelNode operation, final ModelNode model) throws OperationFailedException {
```

Comment:
Please get rid of this. It's not used, isn't documented, and the implementation doesn't work the same way as other abstract OSHs in the package that use the same method names, e.g. performRuntime here is called in Stage.MODEL.